pkg/es/config/config.go

```diff
@@ -33,48 +33,60 @@ import (
 	storageMetrics "github.com/jaegertracing/jaeger/storage/spanstore/metrics"
 )
 
+type TemplateOptions struct {
+	Priority    int `mapstructure:"priority"`
+	NumShards   int `mapstructure:"num_shards"`
+	NumReplicas int `mapstructure:"num_replicas"`
+}
+
+// IndexOptions describes the index format and rollover frequency
+type IndexOptions struct {
+	DateLayout          string `mapstructure:"date_layout"`
+	RolloverFrequency   string `mapstructure:"rollover_frequency"` // "hour" or "day"
+	TemplatePriority    int64  `mapstructure:"template_priority"`
```

Comment:
Why did you change the types to int64? These settings never need that high precision, and the change is forcing you to make other changes.